bootstrap.go

```diff
@@ -96,8 +96,26 @@ func startProfiler(options *options) error {
 }
 
 // stop with a graceful shutdown that includes flushing signals.
-func stop() error {
-	tracer.Stop()
-	profiler.Stop()
-	return nil
+func stop(options *options) error {
+	ctx := context.Background()
+
+	if options.stopTimeout != nil {
```

Comment:
Zero-value timeout I often considered as "no-timeout" in the standard lib. I think following this convetion could simplify the code a bit.